tests/backup/backup_helper.go

```diff
@@ -3447,3 +3465,80 @@ func GetVolumeMounts(AppContextsMapping *scheduler.Context) ([]string, error) {
 	}
 	return nil, fmt.Errorf("unable to find the mount point for %s", AppContextsMapping.App.Key)
 }
+
+type BackupTypeForCSI string
+
+const (
+	NativeCSIWithOffloadToS3 BackupTypeForCSI = "csi_offload_s3"
+	NativeCSI                BackupTypeForCSI = "native_csi"
+	DirectKDMP               BackupTypeForCSI = "direct_kdmp"
+)
+
+// AdditionalBackupRequestParams decorates the backupRequest with additional parameters required
+// when BACKUP_TYPE is Native CSI, Direct KDMP or CSI snapshot with offload to S3
+func AdditionalBackupRequestParams(backupRequest *api.BackupCreateRequest) error {
+	switch strings.ToLower(os.Getenv("BACKUP_TYPE")) {
+	case string(NativeCSIWithOffloadToS3):
+		log.Infof("Detected backup type - %s", NativeCSIWithOffloadToS3)
+		backupRequest.BackupType = api.BackupCreateRequest_Generic
+		var csiSnapshotClassName string
+		var err error
+		if csiSnapshotClassName, err = GetCsiSnapshotClassName(); err != nil {
+			return err
+		}
+		backupRequest.CsiSnapshotClassName = csiSnapshotClassName
+		return nil
+	case string(NativeCSI):
+		log.Infof("Detected backup type - %s", NativeCSI)
+		backupRequest.BackupType = api.BackupCreateRequest_Normal
+		var csiSnapshotClassName string
+		var err error
+		if csiSnapshotClassName, err = GetCsiSnapshotClassName(); err != nil {
+			return err
+		}
+		backupRequest.CsiSnapshotClassName = csiSnapshotClassName
+		return nil
```

Comment:
you can do one return at the end as in golang fall through is not default

Reply:
@lsrinivas-pure But I need to get the `csiSnapshotClassName` string, assign it and only then return from the function.